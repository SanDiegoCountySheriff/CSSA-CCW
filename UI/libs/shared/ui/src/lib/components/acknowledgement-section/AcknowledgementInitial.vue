<template>
  <v-container>
    <v-row>
      <v-col class="text-left acknowledgement-text">
        {{ $t('acknowledgement-initial-part1') }}
      </v-col>
    </v-row>
    <v-row>
      <v-col class="text-left acknowledgement-text">
        {{ $t('acknowledgement-initial-part2') }}
      </v-col>
    </v-row>
    <v-row>
      <v-col class="text-left acknowledgement-text">
        {{ $t('acknowledgement-initial-part3') }}
      </v-col>
    </v-row>
    <v-row>
      <v-col class="text-left acknowledgement-text">
        {{ $t('acknowledgement-initial-part4') }}
      </v-col>
    </v-row>

    <v-row>
      <v-col class="text-left acknowledgement-text">
        {{ $t('acknowledgement-initial-part5') }}
      </v-col>
    </v-row>

    <div class="button-container">
      <AcknowledgementButtonContainer
        @accept="handleAccept"
        @decline="handleDecline"
      />
    </div>
  </v-container>
</template>

<script setup lang="ts">
import AcknowledgementButtonContainer from '@shared-ui/components/containers/AcknowledgementButtonContainer.vue';

interface IAcknowledgementInitialProps {
  handleAccept: CallableFunction;
  handleDecline: CallableFunction;
}

const props = defineProps<IAcknowledgementInitialProps>();
</script>

<style lang="scss" scoped>
.button-container {
  margin: 1.5rem;
}

.acknowledgement-text {
  font-size: 1.3rem;
}
</style>
