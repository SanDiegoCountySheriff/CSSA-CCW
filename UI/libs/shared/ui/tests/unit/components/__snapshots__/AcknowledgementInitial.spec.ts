import Vuetify from 'vuetify';
import AcknowledgementInitial from '@shared-ui/components/acknowledgement-section/AcknowledgementInitial.vue';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe(' AcknowledgementInitial', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    //@ts-ignore
    wrapper = mount(AcknowledgementInitial, {
      localVue,
      vuetify,
      mocks: {
        $t: text => {
          text;
        },
      },
      propsData: {
        handleAccept: () => null,
        handleDecline: () => null,
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
