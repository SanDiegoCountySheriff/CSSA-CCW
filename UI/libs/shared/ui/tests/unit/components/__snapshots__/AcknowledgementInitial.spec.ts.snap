// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` AcknowledgementInitial Should match the snapshot 1`] = `
<div class="container">
  <div class="row">
    <div class="text-left acknowledgement-text col">

    </div>
  </div>
  <div class="row">
    <div class="text-left acknowledgement-text col">

    </div>
  </div>
  <div class="row">
    <div class="text-left acknowledgement-text col">

    </div>
  </div>
  <div class="row">
    <div class="text-left acknowledgement-text col">

    </div>
  </div>
  <div class="row">
    <div class="text-left acknowledgement-text col">

    </div>
  </div>
  <div class="button-container">
    <div class="acknowledgement-button-container"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default success mr-3"><span class="v-btn__content">

  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-3"><span class="v-btn__content">

  </span></button></div>
  </div>
</div>
`;
